### PR TITLE
Feature/request

### PR DESCRIPTION
**Pull request by Snugug**
Add Request

---

`DCO 1.1 Signed-off-by: Sam Richard <sam@snug.ug>`

### CODE REVIEW

**Review by joshblack — APPROVED** [review]
Looks good to me 😄

Might be worth having mocks/stub files in the future for the items.

**Comment by Snugug** [comment]
Agreed we can look at that in the future. @scottnath if this looks good to you, please merge!

File: tests/request.js [hunk 1]
```diff
@@ -1,36 +1,97 @@
...
   return request({
     get: 'all',
     limit: 10,
     punchcard: 'https://punchcard.io',
   }).then(result => {
-    t.deepEqual(result, expected);
+    t.deepEqual(result, all);
   }).catch(e => {
     console.log(e.message); // eslint-disable-line no-console
     t.fail();
```
Comment (scottnath) [inline tests/request.js:75]: [fail](https://github.com/avajs/ava#failmessage) accepts a message argument. Please put `e.message` as a parameter in `t.fail` so if we get this working we can ditch the `console.log`

**Review by scottnath — CHANGES_REQUESTED** [review]
@Snugug no real changes except the one - which isn't even from your new stuff.

File: tests/request.js (hunk 1 above)
Comment (Snugug) [reply tests/request.js:75]: Once we get that working, I will need to come back through and remove `console.log` anyway; why not just wait on that?

Comment (scottnath) [reply tests/request.js:75]: sounds good.

**Review by scottnath — APPROVED** [review]

**Pull request merged**